src/dict/utils/indexedzip.cc

```diff
@@ -108,8 +106,10 @@ bool IndexedZip::loadFile( uint32_t offset, vector< char > & data )
         return false;
       }
 
-      if ( inflate( &stream, Z_FINISH ) != Z_STREAM_END ) {
-        qDebug( "Not zstream end!" );
+      int ret = inflate( &stream, Z_FINISH );
```

Comment:
The `infalteEnd` a few lines later doesn't check `inflateEnd( &stream) != Z_OK`.

Reply:
it is neccessary to check the inflateEnd result?

Reply:
Based on the source code, it is some sanity checks, one of them appears to be checking `z_stream_s->state`.

I am not 100% certain if it is needed because `inflate` could return `Z_STREAM_ERROR` and various other errors.

I think we should check it religiously to make sure all errors are handled.

https://github.com/madler/zlib/blob/ef24c4c7502169f016dcd2a26923dbaf3216748c/inflate.c#L1266-L1276

https://github.com/madler/zlib/blob/ef24c4c7502169f016dcd2a26923dbaf3216748c/inflate.c#L94-L104